spec/models/order_spec.rb

```diff
@@ -0,0 +1,51 @@
+require "rails_helper"
+
+RSpec.describe Order, type: :model do
+  let!(:order) {FactoryBot.create :order}
+  let!(:order_item) {FactoryBot.create :order_item, order_id: order.id}
+  let!(:user) {FactoryBot.create :user}
+
+  describe Order do
+    describe "Associations" do
+      it {expect(order).to have_many(:order_items)}
+      it {expect(order).to belong_to(:user)}
+    end
+  end
+
+  context "Scopes" do
+    describe ".by_status" do
+      it "with status is pending" do
+        expect(Order.by_status(Order.statuses[:pending]).first.status).to eq("pending")
+      end
+      it "with status is not pending" do
+        expect(Order.by_status(Order.statuses[:pending]).first.status).not_to eq("checked")
+      end
+    end
+    describe ".search_by_id" do
+      it "when search order existed" do
+        expect(Order.search_by_id(1).first).to eq(order)
```

Comment:
k fix cứng search như vậy e
e truyền vào 
expect(Order.search_by_id(order.id).first).to eq(order.id)

Reply:
dạ vâng.